server/server_test.go

```diff
@@ -2311,6 +2312,48 @@ func TestPoliciesPathSlashes(t *testing.T) {
 	}
 }
 
+func TestPoliciesUrlEncoded(t *testing.T) {
+	const expectedPolicyID = "/a policy/another-component"
+	var urlEscapedPolicyID = url.PathEscape(expectedPolicyID)
+	f := newFixture(t)
+
+	// PUT policy with URL encoded ID
+	put := newReqV1(http.MethodPut, fmt.Sprintf("/policies/%s", urlEscapedPolicyID), testMod)
+	f.server.Handler.ServeHTTP(f.recorder, put)
+
+	if f.recorder.Code != 200 {
+		t.Fatalf("Expected success but got %v", f.recorder)
+	}
+
+	// end PUT policy with URL encoded ID
+	f.reset()
+	// GET policy with URL encoded ID
+
+	get := newReqV1(http.MethodGet, "/policies", "")
+	f.server.Handler.ServeHTTP(f.recorder, get)
+	if f.recorder.Code != 200 {
+		t.Fatalf("Expected success but got %v", f.recorder)
+	}
+	var listResponse types.PolicyListResponseV1
+	if err := json.NewDecoder(f.recorder.Body).Decode(&listResponse); err != nil {
+		t.Fatalf("Unexpected unmarshal error: %v", err)
+	}
+
+	if listResponse.Result[0].ID != expectedPolicyID {
```

Comment:
OK, so I guess that was meant so since you're extracting the ID from a list response here, but I'm not 100% sure I follow why :)

Reply:
There not a particular reason, that's just what I did 🤷‍♂️
I'll change it to fetch that specific policy later today

Reply:
ok, "today" is a different day, but it is done